roles/create_mce_crs/tasks/main.yml

```diff
@@ -0,0 +1,193 @@
+- name: Create Namespace for imported cluster
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: "{{ namespace }}"
+
+- name: Create Pull Secret
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: v1
+      data:
+        .dockerconfigjson: "{{ dockerconfig }}"
+      kind: Secret
+      metadata:
+        name: pull-secret
+        namespace: "{{ namespace }}"
+      type: kubernetes.io/dockerconfigjson
+
+- name: Create InfraEnv
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: agent-install.openshift.io/v1beta1
+      kind: InfraEnv
+      metadata:
+        name: "{{ namespace }}-infraenv"
+        namespace: "{{ namespace }}"
+        labels:
+          agentclusterinstalls.extensions.hive.openshift.io/location: Dallas
+          networkType: static
+      spec:
+        pullSecretRef:
+          name: pull-secret
+        sshAuthorizedKey: '{{ sshkey }}'
+        nmStateConfigLabelSelector:
+          matchLabels:
+            infraenvs.agent-install.openshift.io: "{{ namespace }}"
+
+- name: Create Cluster Image Set
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: hive.openshift.io/v1
+      kind: ClusterImageSet
+      metadata:
+        name: "openshift-v{{ openshift_full_version }}"
+      spec:
+        releaseImage: "quay.io/openshift-release-dev/ocp-release:{{ openshift_full_version }}-x86_64"
+
+- name: Create Agent Cluster Install
+  kubernetes.core.k8s:
+    state: present
+    kubeconfig: "{{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"
+    definition:
+      apiVersion: extensions.hive.openshift.io/v1beta1
+      kind: AgentClusterInstall
+      metadata:
+        name: "{{ namespace }}"
+        namespace: "{{ namespace }}"
+      spec:
+        apiVIP: "{{ api_vip }}"
+        clusterDeploymentRef:
+          name: "{{ namespace }}"
+        imageSetRef:
+          name: "openshift-v{{ openshift_full_version }}"
+        ingressVIP: "{{ ingress_vip }}"
+        platformType: "{{ platform_type }}"
+        networking:
+          clusterNetwork:
+          - cidr: "{{ cluster_network_cidr }}"
+            hostPrefix: 23
+          serviceNetwork:
+          - "{{ service_network_cidr }}"
+        provisionRequirements:
+          controlPlaneAgents: 3
+        sshPublicKey: "{{ sshkey }}"
+
+- name: Create Kubeconfig
+  command: "oc -n {{ namespace }} create secret generic {{ namespace }}-admin-kubeconfig --from-file=kubeconfig={{kubeconfig_dest_dir}}{{kubeconfig_dest_filename}}"

```

Comment:
I can't find a default so we need a check that `namespace` is defined.

Reply:
Added a check in the inventory validation